dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java

```diff
@@ -60,17 +60,22 @@ public Contentlet populateContentletFromMap(final Contentlet contentlet,
      * @param contentlet {@link Contentlet} original contentlet to hydrate, won't be modified.
      * @return Contentlet returns a contentlet, if there is something to add will create a new instance based on the current one in the parameter and the new attributes, otherwise will the same instance
      */
-    public Contentlet hydrateContentLet (final Contentlet contentlet) {
+    public Contentlet hydrateContentLet(final Contentlet contentlet) {
 
         Contentlet newContentlet = contentlet;
-        if (null != contentlet && !contentlet.getMap().containsKey(HTMLPageAssetAPI.URL_FIELD)) {
-
-            final String url = this.getUrl(contentlet);
-            if (null != url) {
-                // making a copy to avoid issues on modifying cache objects.
-                newContentlet = new Contentlet();
-                newContentlet.getMap().putAll(contentlet.getMap());
-                newContentlet.getMap().put(HTMLPageAssetAPI.URL_FIELD, url);
+
+        if (null != contentlet) {
+            // making a copy to avoid issues on modifying cache objects.
```

Comment:
I would do this more generic in order to remove not only the null properties but eventually any other not serializable property

Reply:
done